### ISSUE TITLE
Autosizing cells using Auto-layout with Supplementary Views

### ISSUE THREAD

**Issue by zachwhelchel** [Instagram/IGListKit#508]
Currently in iOS 10 you can't use self sizing cells (estimatedItemSize & preferredLayoutSizeFittingSize) when you have supplementary views in a collection view. As seen here:

http://stackoverflow.com/questions/40129245/ios-10-uicollectionviewflowlayout-using-uicollectionviewflowlayoutautomaticsize

To work around this in the past I've been keeping around a "dummy" cell off screen: one per type, in order to get the proper sizing. Using IGListKit though, I'm not sure how I'd go about finding a workaround. I guess I could pass the "dummy" cell into the IGListSectionController along with the object by grouping them together in a single object, but that is very messy.

Any other thoughts? I guess I might have to ditch using auto-layout to figure out the height of a cell, but I would really love to keep auto-layout as the truth source for cell height.

In summary; this isn't an issue with IGListKit, but rather with UICollectionView, that poses a difficulty when trying to implement a fix while using IGListKit.

**Comment by rnystrom** [comment]
@zachwhelchel ugh, ya, that's a tough one. Estimated cells in general have been giving us headaches. 

IMO passing in a cell for each section controller is fine since that's basically the same solution you'd use w/out IGLK, right?

Would be interesting if someone cooked up a section controller that did all this for you, though. I think it'd be possible: when dequeueing a cell, create a dummy one and use that in `sizeForItemAtIndex:` or something.

Highly likely that we wont do too much with it, though, since Instagram doesn't use self-sizing-cells or auto layout at all, perf will be tricky, and its really opening a can of worms 😬

**Comment by jessesquires** [comment]
Closing to track at #516
